components/binding/internal/worker/targetkind.go

```diff
@@ -87,7 +71,29 @@ func (w *TargetKindWorker) RemoveProcess(targetKind *v1alpha1.TargetKind, log lo
 	return w.storage.Unregister(*targetKind)
 }
 
-func (w *TargetKindWorker) isDifferentThanRegistered(targetKind *v1alpha1.TargetKind) (bool, error) {
+func (w *TargetKindWorker) register(targetKind *v1alpha1.TargetKind, status string) error {
+	err := w.storage.Register(*targetKind)
+	if err != nil {
+		return errors.Wrapf(err, "while registering TargetKind %q", targetKind.Name)
+	}
+	switch status {
+	case v1alpha1.TargetKindRegistered:
+		err = targetKind.Status.Registered()
```

Comment:
we also have a message field in `.status`, it should inform why TK is failed

Reply:
added message on fail: 

```
Failed because a TargetKind with the same kind and different properties exists
```